### PR TITLE
Update deploy_infrastructure.sh

### PR DESCRIPTION
**Pull request by LiliamLeme**
Az ad reset ID in case the password has na hyphen

# Type of PR
- Code changes

- Documentation changes
- Code changes
- Test changes
- CI-CD changes
- GitHub Template changes

## Purpose

- ...

## Author pre-publish checklist

- [X ] Added test to prove my fix is effective or new feature works
- [ ] No PII in logs
- [ ] Made corresponding changes to the documentation

## Validation steps
 TESTs
- ...
++ local sprincipal=223XXX
++ [[ -z -gb8CCC --> persisted password with hyphen 
++ [[ - == \- ]]
++ local sp_stor_outval
+++ az ad sp credential reset 223XXX
+ password=pZ3 --> recreated 
====================================
test:

#!/bin/bash
#!/bin/bash

validate_password() {
    local password="$1"
    local sprincipal="$2"

   while [[ -z "$password" || "${password:0:1}" == "-" ]]; do
      local sp_stor_outval
      sp_stor_outval=$(az ad sp credential reset --id "$sprincipal" 2>/dev/null)
         
      if [[ $? -ne 0 ]]; then
            echo "Error: Failed to reset credentials for Service Principal: $sprincipal" >&2
            password=""  
            break  
      fi
      password=$(echo "$sp_stor_outval" | jq -r '.password')
   done
   echo "$password"
}

# Example variables
sp_stor_pass="-ABxx"
sp_stor_id="43fxx"

# Validate and update the password
sp_stor_pass=$(validate_password "$sp_stor_pass" "$sp_stor_id")

if [[ -z "$sp_stor_pass" ]]; then
    echo "Warning: Unable to validate or reset the password for Service Principal." >&2
    # Handle this case gracefully without exiting the script
    # For example, you can log or skip actions that require this password
else
 
    echo "Validated Password:"
    echo "$sp_stor_pass"
fi

result:

Validated Password:
SAJ8xxx

====================================
## Issues Closed or Referenced

- Closes #issue_number

- References #issue_number

### CODE REVIEW

File: e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh [hunk 1]
```diff
@@ -178,17 +178,19 @@ az keyvault secret set --vault-name "$kv_name" --name "applicationInsightsConnec
...
     local password="$1"
+    local sprincipal="$2"
    
     if [[ -z "$password" || "${password:0:1}" == "-" ]]; then
-        password="R${password:1}"
-        echo "$password"
-     
+       local sp_stor_outval
+       sp_stor_outval=$(az ad sp credential reset --id "$sprincipal" )
+       password=$(echo "$sp_stor_outval" | jq -r '.password')
```
Comment (ydaponte) [inline e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh:190]: It is unlikely, BUT what happens if the reset password starts with "-" as well? Can we recursively call the validate password on the new generated password?

**Review by ydaponte — CHANGES_REQUESTED** [review]
There is a scenario where it would not work - it is unlikely but can happen that the re-generated password also starts with an "-"? We might need to recursively call the validate_password() on the reset pwd

File: e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh (hunk 1 above)
Comment (LiliamLeme) [reply e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh:190]: Yeah... true... But I thought 2X would be unlikely.. that is why I did not added the recursive, so I did not want to make it more complicate... even once it is quite  random right>

File: e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh [hunk 2]
```diff
@@ -312,7 +318,8 @@ sp_adf_pass=$(echo "$sp_adf_out" | jq -r '.password')
 sp_adf_tenant=$(echo "$sp_adf_out" | jq -r '.tenant')
 
 # Validate the password
-sp_adf_pass=$(validate_password "$sp_adf_pass")
+sp_adf_pass=$(validate_password "$sp_stor_pass" "$sp_stor_id" )
```
Comment (elenaterenzi) [inline e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh:321]: you are using the wrong variables here, you are still referencing to the storage variables, you should refer to the ADF variables instead

File: e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh [hunk 3]
```diff
@@ -178,18 +178,25 @@ az keyvault secret set --vault-name "$kv_name" --name "applicationInsightsConnec
 # ###########################
 echo "validate_password function..."
 # ###########################
-###if there is an hyphen in the first position replace by R
+###if there is an hyphen reset until valid password in found in case of error empty the pwd.
 # ###########################
+
 validate_password() {
```
Comment (elenaterenzi) [inline e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh:184]: I would suggest moving this function at the beginning of the script for better code readability

**Review by elenaterenzi — CHANGES_REQUESTED** [review]

File: e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh (hunk 3 above)
Comment (LiliamLeme) [reply e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh]: done

**Review by ydaponte — APPROVED** [review]
After all the solution was far much more simple! LGTM!

**Comment by LiliamLeme** [comment]
merged

**Pull request merged**